services/spar/test/Test/Spar/Scim/UserSpec.hs

```diff
@@ -0,0 +1,189 @@
+type InterpreterState =
+  ( Map (Data.Id.TeamId, Wire.API.User.Identity.Email) Data.Id.UserId,
+    ( Map Data.Id.UserId (Data.Json.Util.UTCTimeMillis, Data.Json.Util.UTCTimeMillis),
+      ( Map UserRefOrd UserId,
+        (Spar.Sem.IdPConfigStore.Mem.TypedState, Either ScimError ())
+      )
+    )
+  )
+
+handlerResult :: InterpreterState -> Either ScimError ()
+handlerResult = snd . snd . snd . snd
+
+interpretWithBrigAccessMock ::
+  ( Sem Effs (Either ScimError ()) ->
+    Sem EffsWithoutBrigAccess (Either ScimError ())
+  ) ->
+  Sem Effs (Either ScimError ()) ->
+  IO InterpreterState
+interpretWithBrigAccessMock mock =
+  runFinal
+    . embedToFinal @IO
+    . discardTinyLogs
+    . scimExternalIdStoreToMem
+    . scimUserTimesStoreToMem
+    . samlUserStoreToMem
+    . idPToMem
+    . mock
+
+mockBrigForNonExistendUser ::
+  forall (r :: EffectRow).
+  Members '[Embed IO] r =>
+  Sem (BrigAccess ': r) (Either ScimError ()) ->
+  Sem r (Either ScimError ())
+mockBrigForNonExistendUser = interpret $ \case
+  (GetAccount WithPendingInvitations _) -> pure Nothing
+  (Spar.Sem.BrigAccess.DeleteUser _) -> pure NoUser
+  _ -> do
+    liftIO $ expectationFailure $ "Unexpected effect (call to brig)"
+    error "Throw error here to avoid implementation of all cases."
+
+mockBrigForPartiallyDeletedUser ::
+  forall (r :: EffectRow).
+  Members '[Embed IO] r =>
+  Sem (BrigAccess ': r) (Either ScimError ()) ->
+  Sem r (Either ScimError ())
+mockBrigForPartiallyDeletedUser = interpret $ \case
+  (GetAccount WithPendingInvitations _) -> pure Nothing
+  (Spar.Sem.BrigAccess.DeleteUser _) -> pure AccountDeleted
+  _ -> do
+    liftIO $ expectationFailure $ "Unexpected effect (call to brig)"
+    error "Throw error here to avoid implementation of all cases."
+
+mockBrigForActiveUser ::
+  forall (r :: EffectRow).
+  Members '[Embed IO] r =>
+  UserAccount ->
+  DeleteUserResult ->
+  Sem (BrigAccess ': r) (Either ScimError ()) ->
+  Sem r (Either ScimError ())
+mockBrigForActiveUser acc deletionResult = interpret $ \case
+  (GetAccount WithPendingInvitations uid) ->
+    if uid == (userId . accountUser) acc
+      then pure $ Just acc
+      else pure Nothing
+  (Spar.Sem.BrigAccess.DeleteUser _) -> pure deletionResult
+  _ -> do
+    liftIO $ expectationFailure $ "Unexpected effect (call to brig)"
+    error "Throw error here to avoid implementation of all cases."
+
+someActiveUser :: ScimTokenInfo -> IO UserAccount
+someActiveUser tokenInfo = do
+  user <- generate arbitrary
+  pure $
+    UserAccount
+      { accountStatus = Active,
+        accountUser =
+          user
+            { userDisplayName = Name "Some User",
+              userAccentId = defaultAccentId,
+              userPict = noPict,
+              userAssets = [],
+              userHandle = parseHandle "some-handle",
+              userIdentity = (Just . EmailIdentity . fromJust . parseEmail) "someone@wire.com",
+              userTeam = Just $ stiTeam tokenInfo
+            }
```

Comment:
why not just `user`?  why do you update all these fields?  i can't find anything in this module that expects these to be anything in particular.

Reply:
I thought it would be good to provide a valid / sound / half-way realistic test object: It's easier to extend or add tests when the test data isn't too artificial.